test/python/quantum_info/operators/test_dihedral.py

```diff
@@ -44,8 +47,25 @@ def random_cnotdihedral_circuit(num_qubits, num_gates, gates="all", seed=None):
     if gates == "all":
         if num_qubits == 1:
             gates = ["i", "x", "y", "z", "t", "tdg", "s", "sdg"]
+        elif num_qubits < 3:
```

Comment:
Nitpick and just a personal preference.
```python
gates = ["i", "x", "y", "z", "t", "tdg", "s", "sdg"]

if num_qubits >= 2:
    gates += ["cx", "cz", "cs", "csdg", "swap"]
if num_qubits >= 3:
    gates += ["ccz"]
```
would make added element clearer (and prevent to forget adding new elements to higher qubits gate set). It's up to you.